src/pydocstyle/checker.py

```diff
@@ -682,16 +682,23 @@ def _check_args_section(docstring, definition, context):
                 except `self` or `cls` if it is a method.
 
         """
+        docstring_args = set()
+        for line in context.following_lines:
+            match = ConventionChecker.GOOGLE_ARGS_REGEX.match(line)
+            if match:
+                docstring_args.add(match.group(1))
         if definition.kind == 'function':
             function_pos_args = get_function_args(definition.source)
-            docstring_args = set()
-            for line in context.following_lines:
-                match = ConventionChecker.GOOGLE_ARGS_REGEX.match(line)
-                if match:
-                    docstring_args.add(match.group(1))
-            missing_args = function_pos_args - docstring_args
-            if missing_args:
-                yield violations.D417(", ".join(missing_args), definition.name)
+        elif definition.kind == 'method':
+            isnt_static = all(map(lambda decorator: decorator.name != 'staticmethod',
+                              definition.decorators))
+            # If the method isn't static, then we skip the first positional argument as it
+            # is `cls` or `self`
+            function_pos_args = get_function_args(definition.source, skip_first_arg=isnt_static)
+        missing_args = function_pos_args - docstring_args
+        if missing_args:
+            yield violations.D417(", ".join(sorted(missing_args)), definition.kind, definition.name)
```

Comment:
@Nurdok because it looks like we just call `pycodestyle --help`

https://github.com/PyCQA/pydocstyle/blame/master/tox.ini#L27

Reply:
I have fixed this instance, but it might be worth running `black` on top of the code since I noticed a lot of pycodestyle violations all over the place.